src/main/scala/mesosphere/marathon/api/serialization/ContainerSerializer.scala

```diff
@@ -152,12 +152,15 @@ object PortMappingSerializer {
       proto.getLabelsList.asScala.map { p => p.getKey -> p.getValue }.toMap
     )
 
-  def toMesos(mapping: Container.Docker.PortMapping): mesos.Protos.ContainerInfo.DockerInfo.PortMapping = {
-    mesos.Protos.ContainerInfo.DockerInfo.PortMapping.newBuilder
-      .setContainerPort(mapping.containerPort)
-      .setHostPort(mapping.hostPort)
-      .setProtocol(mapping.protocol)
-      .build
+  def toMesos(mapping: Container.Docker.PortMapping): Seq[mesos.Protos.ContainerInfo.DockerInfo.PortMapping] = {
+    def mesosPort(protocol: String) = {
+      mesos.Protos.ContainerInfo.DockerInfo.PortMapping.newBuilder
+        .setContainerPort (mapping.containerPort)
+        .setHostPort(mapping.hostPort)
+        .setProtocol(protocol)
+        .build
+    }
+    mapping.protocol.split(',').toSet.map(mesosPort).toList
```

Comment:
Is `toSet` needed? In the validation you make sure, that the protocols are distinct.